api/internal/core/store/validate.go

```diff
@@ -238,22 +238,37 @@ func (v *APISIXJsonSchemaValidator) Validate(obj interface{}) error {
 		return err
 	}
 
+	pluginDisableSchema := map[string]map[string]interface{}{
+		"disable": {"type": "boolean"},
+	}
 	plugins, schemaType := getPlugins(obj)
-	//fix lua json.encode transform lua{properties={}} to json{"properties":[]}
-	reg := regexp.MustCompile(`\"properties\":\[\]`)
 	for pluginName, pluginConf := range plugins {
-		var schemaDef string
-		schemaDef = conf.Schema.Get("plugins." + pluginName + "." + schemaType).String()
-		if schemaDef == "" && schemaType == "consumer_schema" {
-			schemaDef = conf.Schema.Get("plugins." + pluginName + ".schema").String()
+		schemaValue := conf.Schema.Get("plugins." + pluginName + "." + schemaType).Value()
+		if schemaValue == nil && schemaType == "consumer_schema" {
+			schemaValue = conf.Schema.Get("plugins." + pluginName + ".schema").Value()
 		}
-		if schemaDef == "" {
-			log.Warnf("schema validate failed: schema not found, path: %s", "plugins."+pluginName)
+		if schemaValue == nil {
+			log.Warnf("schema validate failed: schema not found,  %s, %s", "plugins."+pluginName, schemaType)
 			return fmt.Errorf("schema validate failed: schema not found, path: %s", "plugins."+pluginName)
 		}
+		schemaMap := schemaValue.(map[string]interface{})
+		if schemaMap["type"] != nil && schemaMap["type"].(string) == "object" {
+			if properties, ok := schemaMap["properties"]; ok {
+				propertiesMap := properties.(map[string]interface{})
+				if len(propertiesMap) == 0 {
+					schemaMap["properties"] = pluginDisableSchema
+				}
+			} else {
+				schemaMap["properties"] = pluginDisableSchema
+			}
+		}
+		schemaByte, err := json.Marshal(schemaMap)
+		if err != nil {
+			log.Warnf("scheme validate failed: schema not found, path: %s, %w", "plugins."+pluginName, err)
+			return fmt.Errorf("scheme validate failed: schema not found, path: %s, %w", "plugins."+pluginName, err)
```

Comment:
error info is wrong, here is a json.Marshal action.

Reply:
updated.